pkg/apis/apps/validation/validation_test.go

```diff
@@ -62,53 +62,104 @@ func makeSet(selectorLabels *map[string]string, template *api.PodTemplate, tweak
 	return s
 }
 
-func tweakName(name string) setTweak {
-	return func(s apps.StatefulSet) {
-		s.ObjectMeta.Name = name
+func tweakName(name string) statefulSetTweak {
+	return func(ss apps.StatefulSet) {
+		ss.ObjectMeta.Name = name
 	}
 }
 
-func tweakNamespace(ns string) setTweak {
-	return func(s apps.StatefulSet) {
-		s.ObjectMeta.Namespace = ns
+func tweakNamespace(ns string) statefulSetTweak {
+	return func(ss apps.StatefulSet) {
+		ss.ObjectMeta.Namespace = ns
 	}
 }
 
-func tweakManagementPolicy(policy apps.PodManagementPolicyType) setTweak {
-	return func(s apps.StatefulSet) {
-		s.Spec.PodManagementPolicy = policy
+func tweakMetaLabels(labels map[string]string) statefulSetTweak {
+	return func(ss apps.StatefulSet) {
+		ss.ObjectMeta.Labels = labels
 	}
 }
 
-func tweakRepicas(replicas int32) setTweak {
-	return func(s apps.StatefulSet) {
-		s.Spec.Replicas = replicas
+func tweakAnnotations(annotations map[string]string) statefulSetTweak {
+	return func(ss apps.StatefulSet) {
+		ss.ObjectMeta.Annotations = annotations
 	}
 }
 
-func tweakPodManagementPolicy(policy apps.PodManagementPolicyType) setTweak {
-	return func(s apps.StatefulSet) {
-		s.Spec.PodManagementPolicy = policy
+func tweakFinalizers(finalizers []string) statefulSetTweak {
```

Comment:
Optional:

For finalizers, labels, annotations, you accept maps or slices, which makes the caller have to specify them.  If the goal is to make the callers as obvious and readable as possible, there's another way to approach this.  YOu could do:

```
func tweakFinalizers(finalizers ...string) statefulSetTweak {
```

This let's the caller say `tweakFinalizers("finalizer1")` or, if they really need more than one, `tweakFinalizers("one", "two", "three")`.

For the maps, you could do either:

```
func tweakLabels(key, value) statefulSetTweak {
    return func(ss apps.StatefulSet) {
        if ss.Labels == nil {
            ss.Labels = map[string]string{}
        }
        ss.Labels[key] = value
    }
}
```

If we really need to set more than one, you can call this more than one time.  Or you can do something like:

```
func tweakMetaLabels(keysAndValues ...string) statefulSetTweak {
```

But this is sort of awkward, IMO

Reply:
modified as suggested.

However, as the selector labels are typically already specified, I find it maybe more convenient to continue accepting a map in the tweak function.

Reply:
OK - the goal is "as obvious as possible, and not one step further".  If you think that's the right tradeoff, go for it.